zppy/templates/readTS.py

```diff
@@ -62,10 +69,15 @@ def globalAnnual(self, var):
             # Non-derived variables
 
             # Read variable
-            v = self.f(var)
+            v = self.f.data_vars[var]
+            # Refactor note: `v = self.f(var)` gave `cdms2.tvariable.TransientVariable`
             units = v.units
 
             # Annual average
-            v = cdutil.YEAR(v)
+
+            # Refactor note: `AttributeError: 'Dataset' object has no attribute 'temporal'` seems to always occur
+            # Regardless if using CDAT or not, if using as object or class method.
+            # v = self.f.temporal.group_average(v, "year")
+            v = xarray.Dataset.temporal.group_average(v, "year")
```

Comment:
@tomvothecoder It does indeed seem like I'm able to eliminate the `cdscan` call to generate an `xml` + the `self.f = cdms2.open(filename)` line by using `xarray.open_mfdataset(f"{directory}/*.nc")`

However, no matter what I do, Python doesn't seem to think `temporal` exists. I'm looking at https://xcdat.readthedocs.io/en/latest/generated/xarray.Dataset.temporal.group_average.html, and this is running in the latest Unified environment. Am I using the wrong averaging function here? (This line replaces `v = cdutil.YEAR(v)`)